src/decoder.rs

```diff
@@ -120,6 +123,33 @@ pub fn strip_junk_header(slice: &[u8]) -> io::Result<&[u8]> {
     Ok(&slice[slice.len()..])
 }
 
+/// Decodes range mappping bitfield string into index
+fn decode_rmi(rmi_str: &str, val: &mut BitVec<u8, Lsb0>) -> Result<()> {
+    val.clear();
+    if val.capacity() < rmi_str.len() * 6 {
+        val.reserve(rmi_str.len() * 6 - val.capacity());
+    }
+    unsafe {
+        // Safety: we just reserved enough space
+        val.set_len(rmi_str.len() * 6);
+    };
```

Comment:
You should just use `val.resize(mri_str.len() * 6, false)` to resize this vec and zero out all the values. This will take care of managing capacity, and avoids unsafe code.